core/controllers/story_editor.py

```diff
@@ -144,14 +155,31 @@ class StoryPublishHandler(base.BaseHandler):
     """A data handler for publishing and unpublishing stories."""
 
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {
+        'story_id': {
+            'schema': {
+                'type': 'basestring'
+            }
+        }
+    }
+    HANDLER_ARGS_SCHEMAS = {
+        'PUT': {
+            'new_story_status_is_public': {
+                'schema': {
+                    'type': 'bool'
+                },
+                'default_value': None
+            }
+        }
+    }
 
     @acl_decorators.can_edit_story
     def put(self, story_id):
         """Published/unpublished given story."""
         story = story_fetchers.get_story_by_id(story_id, strict=False)
         topic_id = story.corresponding_topic_id
 
-        new_story_status_is_public = self.payload.get(
+        new_story_status_is_public = self.normalized_payload.get(
             'new_story_status_is_public')
         if not isinstance(new_story_status_is_public, bool):
             raise self.InvalidInputException
```

Comment:
This is no longer needed, since you introduced the validation in the schema.

Reply:
ditto elsewhere

Reply:
done

Reply:
done